plugins/filter_aws/aws.c

```diff
@@ -627,6 +867,18 @@ static int cb_aws_filter(const void *data, size_t bytes,
             msgpack_pack_str_body(&tmp_pck,
                                   ctx->hostname, ctx->hostname_len);
         }
+
+        if (ctx->tags_include) {
+            for (i = 0; i < ctx->tags_count; i++) {
+                msgpack_pack_str(&tmp_pck, ctx->tag_keys_len[i]);
+                msgpack_pack_str_body(&tmp_pck,
+                                    ctx->tag_keys[i],
+                                    ctx->tag_keys_len[i]);
+                msgpack_pack_str(&tmp_pck, ctx->tag_values_len[i]);
+                msgpack_pack_str_body(&tmp_pck,
+                                    ctx->tag_values[i], ctx->tag_values_len[i]);
+            }
+        }
```

Comment:
About a hundred lines up in this function you will see:

```
/* new record map size is old size + the new keys we will add */
        total_records = obj->via.map.size + ctx->new_keys;
        msgpack_pack_map(&tmp_pck, total_records);
```

The new keys in the map for tags needs to be accounted for.

Reply:
Isn't it enough to add `tags_count` to `new_keys` in `get_ec2_metadata` function?

```
    if (ctx->tags_include && (!ctx->tag_keys || !ctx->tag_values)) {
        ret = get_ec2_tags(ctx);
        if (ret < 0) {
            return -1;
        }
        ctx->new_keys += ctx->tags_count;
    }

```

Reply:
By the way, shouldn't the `ctx->new_keys` be set to zero at the beginning of `get_ec2_metadata`? This function might be called multiple times, therefore `new_keys` might be wrong if called multiple times.

Reply:
> By the way, shouldn't the ctx->new_keys be set to zero at the beginning of get_ec2_metadata? This function might be called multiple times, therefore new_keys could be inflated (e.g. be equal to 2 times higher value than should be).

Yea... same as how you free and realloc the memory for the tags...

> Isn't it enough to add tags_count to new_keys in get_ec2_metadata function?

Sorry I missed that. You're good.